docs/tutorials/mongoid-transactions.txt

```diff
@@ -0,0 +1,63 @@
+=========
+Sessions
+=========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Versions 4.0 and higher of the MongoDB server support transactions. You can use sessions with Mongoid in a similar way
+that you would execute a transaction in ActiveRecord as long as the current thread is within a session. Namely, you can
+call a method, ``#with_transaction`` on a model class or on an instance of a model from within a session block and execute
+some operations in a new block. All operations in the inner block will be executed in the context of single transactions. Please
+see the MongoDB Ruby driver documentation for what transaction options are available.
+
+Please note the following limitations of sessions:
+
+- Transactions cannot be shared across threads; transactions are not thread-safe. This is consistent with the Ruby driver's support for transactions.
+
+- Transactions cannot be nested. You cannot called ``#with_transaction`` on a model class or a model instance within the block passed to the ``#with_transaction`` method on another model class or model instance.
+
+- Transactions cannot be used outside of a session. You must be within a session block initiated on the same model class or model instance.
+
+- All model classes and instances used within the transaction block must use the same driver client. For example, if you have specified different ``storage_options`` for another model used in the block than that of the model class or instance on which ``#with_session`` is called, you will get an error.
```

Comment:
There needs to be guidance provided to the user as to what to do in these situations.

Reply:
This is copied directly from the sessions tutorial (mod changing the word "session" to "session"; I'm not sure what else to say beyond "use the same client", which is already stated.

Reply:
I think @johnnyshields 's comment is in relation to this. What is an example of Mongoid models and configuration which will cause a user to "get an error" as described here?

Reply:
As the first sentence states, if you use a model class or instance that has a different underlying client, you'll get an error.

Reply:
I doubt a user reading this would know how the underlying clients are constructed without a link to an explanation or an example.

Reply:
Considering that all models will use the same client unless explicitly configured not to, I'm not sure how a user could run into issues with this if they don't know how to change clients in the first place.

Reply:
Exactly. The current wording says a certain usage is disallowed, the doc should make it easy for a user to figure out whether their usage is allowed or not.

Reply:
I added a clarification [here](https://github.com/mongodb/mongoid/pull/4507/commits/cb8bc43b65935459c7a1f8aa916abfc47216eb87); do you find this sufficient?